models/backbone.py

```diff
@@ -64,15 +65,17 @@ def __init__(self, backbone: nn.Module, train_backbone: bool, num_channels: int,
         if return_interm_layers:
             return_layers = {"layer1": "0", "layer2": "1", "layer3": "2", "layer4": "3"}
         else:
-            return_layers = {'layer4': 0}
+            return_layers = {'layer4': "0"}
```

Comment:
This oversight took me a while to figure out, as the error messages were pretty cryptic

<details>
<summary>Error message from torchscript</summary>

```python
Traceback (most recent call last):
  File "test_all.py", line 64, in test_model_script
    scripted_model = torch.jit.script(model)
  File "/Users/fmassa/anaconda3/lib/python3.6/site-packages/torch/jit/__init__.py", line 1340, in script
    return torch.jit._recursive.create_script_module(obj, torch.jit._recursive.infer_methods_to_compile)
  File "/Users/fmassa/anaconda3/lib/python3.6/site-packages/torch/jit/_recursive.py", line 313, in create_script_module
    return create_script_module_impl(nn_module, concrete_type, stubs_fn)
  File "/Users/fmassa/anaconda3/lib/python3.6/site-packages/torch/jit/_recursive.py", line 367, in create_script_module_impl
    script_module = torch.jit.RecursiveScriptModule._construct(cpp_module, init_fn)
  File "/Users/fmassa/anaconda3/lib/python3.6/site-packages/torch/jit/__init__.py", line 1713, in _construct
    init_fn(script_module)
  File "/Users/fmassa/anaconda3/lib/python3.6/site-packages/torch/jit/_recursive.py", line 348, in init_fn
    scripted = create_script_module_impl(orig_value, sub_concrete_type, infer_methods_to_compile)
  File "/Users/fmassa/anaconda3/lib/python3.6/site-packages/torch/jit/_recursive.py", line 367, in create_script_module_impl
    script_module = torch.jit.RecursiveScriptModule._construct(cpp_module, init_fn)
  File "/Users/fmassa/anaconda3/lib/python3.6/site-packages/torch/jit/__init__.py", line 1713, in _construct
    init_fn(script_module)
  File "/Users/fmassa/anaconda3/lib/python3.6/site-packages/torch/jit/_recursive.py", line 348, in init_fn
    scripted = create_script_module_impl(orig_value, sub_concrete_type, infer_methods_to_compile)
  File "/Users/fmassa/anaconda3/lib/python3.6/site-packages/torch/jit/_recursive.py", line 367, in create_script_module_impl
    script_module = torch.jit.RecursiveScriptModule._construct(cpp_module, init_fn)
  File "/Users/fmassa/anaconda3/lib/python3.6/site-packages/torch/jit/__init__.py", line 1713, in _construct
    init_fn(script_module)
  File "/Users/fmassa/anaconda3/lib/python3.6/site-packages/torch/jit/_recursive.py", line 348, in init_fn
    scripted = create_script_module_impl(orig_value, sub_concrete_type, infer_methods_to_compile)
  File "/Users/fmassa/anaconda3/lib/python3.6/site-packages/torch/jit/_recursive.py", line 367, in create_script_module_impl
    script_module = torch.jit.RecursiveScriptModule._construct(cpp_module, init_fn)
  File "/Users/fmassa/anaconda3/lib/python3.6/site-packages/torch/jit/__init__.py", line 1713, in _construct
    init_fn(script_module)
  File "/Users/fmassa/anaconda3/lib/python3.6/site-packages/torch/jit/_recursive.py", line 333, in init_fn
    cpp_module.setattr(name, orig_value)
RuntimeError: Unable to cast Python instance to C++ type (compile in debug mode for details)
```

</details>

cc @eellison if this error message could be improved it would be great.